### ISSUE TITLE
bndtools integration - help needed

### ISSUE THREAD

**Issue by gadieichhorn** [primeval-io/aspecio#1]
Hi,

I am trying to use Aspecio on a bndtools (eclipse) project.
I am using the example from this repo but I am not getting any aspects loaded or weaved

Any suggestions for bndtools users?

```
15:08:57.484 [main] INFO io.lambdacube.aspecio.internal.service.AspecioImpl - Activating Aspecio
15:08:57.484 [main] INFO io.lambdacube.aspecio.internal.service.AspecioImpl - Aspecio activated
```

one of my components is annotated

`@Weave(required = CountingAspect.class)
`
I then try to load the aspects in another of my components

```
	@Reference
	private Aspecio aspecio;

...

      /// later I try to call this service
	Set<String> aspects = aspecio.getRegisteredAspects();
       aspects.stream().forEach(aspect -> LOGGER.info("Aspect: [{}]", aspect));
```

the Set is always empty.

I also tried to call aspect from the gogo shell. no results...

```
g! aspects
g! 
```

Thanks.

**Comment by magnet** [comment]
Hi @gadieichhorn 

First sorry for the delay, I was on holiday break and did not check GitHub much, so I did not see the notification.

The @Weave annotation is a fancy way of adding a component property, and is generated by a plugin you can find here: https://github.com/lambdacube/bnd-dsap-plugin

Have you set it up in your build as described in the documentation? You need to add this to bnd.bnd (or to a shared bnd file) to allow the bnd plugin to process components and add the properties.

```-plugin: io.lambdacube.bnd.component.annotation.properties.DSAPPlugin```


Also, I have not yet tried it with the latest Bnd version. I am not sure if it bnd-dsap-plugin will work, but I'm planning on making sure it does in the upcoming weeks.
If you can't get the annotation to work, you can always declare that property using ```@Component(property = "service.aspect.weave=io.lambdacube.aspecio.examples.aspect.counting.CountingAspect")``` (the property value being the fully qualified name of the CountingAspect class).

Another thing to make sure is that Aspecio is started (like you see with that log message) before your component is activated. I plan to fix that in the coming month as well.

Please let me know if it works :)
Simon

**Comment by gadieichhorn** [comment]
Thanks for this answer.
How do I change the aspecio start level?

**Comment by magnet** [comment]
With bnd launcher, Bundles are already started in the order they are declared in runbundles, according to https://github.com/bndtools/bnd/issues/141
If you use the resolver you must edit to make sure aspecio is earlier in the runbundes list than your application bundles (but after slf4j)

In the next version of Aspecio it will restart bundles that want weaving to make sure it works. It's the only way with current OSGi APIs to make it work regardless of start ordering.

**Comment by magnet** [comment]
Closing this as Aspecio is moving to [Primeval](http://github.com/primeval-io/aspecio). 
See primeval-io/aspecio#3 for the issue tracking build order. A fix is planned very soon but most of all Aspecio 2.0.0 is bringing a completely new proxy/interception model (already available in 2.0.0-SNAPSHOT)
